notification/source/c/src/kaa_demo.c

```diff
@@ -57,9 +59,28 @@ static int read_topic_id(void)
     }
 
     input[input_length] = '\0';
+    if (strcmp(input, "quit") == 0) {
+        exit(EXIT_SUCCESS);
+    }
     return atoi(input);
 }
 
+static kaa_topic_t *find_topic(uint64_t topic_id)
+{
+    if (!topic_list || !kaa_list_get_size(topic_list)) {
+        return NULL;
+    }
+    kaa_list_node_t *it = kaa_list_begin(topic_list);
+    while (it) {
```

Comment:
`kaa_list` already has the necessary API for search:
 `kaa_list_node_t *kaa_list_find_next(kaa_list_node_t *from, match_predicate pred, void *context);`
Refer to `kaa_list.h`